CMakeLists.txt

```
@@ -91,6 +74,56 @@ if(ENABLE_MKLCPU_BACKEND
   list(APPEND DOMAINS_LIST "rng")
 endif()
 
+# Define required CXX compilers before project
+if(CMAKE_CXX_COMPILER)
+  if(WIN32)
+    string(REPLACE "\\" "/" CMAKE_CXX_COMPILER ${CMAKE_CXX_COMPILER})
+  endif()
+else()
+  if(ENABLE_CUBLAS_BACKEND OR ENABLE_CURAND_BACKEND)
+    set(CMAKE_CXX_COMPILER "clang++")
+  elseif(ENABLE_MKLGPU_BACKEND)
+    set(CMAKE_CXX_COMPILER "dpcpp")
+  else()
+    find_program(DPCPP_PATH dpcpp)
+    if(DPCPP_PATH)
+      set(CMAKE_CXX_COMPILER "dpcpp")
+    else()
+      if(WIN32)
+        message(STATUS "dpcpp was not found in PATH, using clang-cl instead")
+        set(CMAKE_CXX_COMPILER "clang-cl")
+      else()
+        message(STATUS "dpcpp was not found in PATH, using clang++ instead")
+        set(CMAKE_CXX_COMPILER "clang++")
+      endif()
+    endif()
+  endif()
+endif()
+
+# Define required C compilers for Windows before project
+if(WIN32)
```

I believe here as well would be preferable to only set the compiler explicitly in case we are using `dpc++` in hipSYCL we make efforts internally to find the proper settings even in case targeting windows

```suggestion
if(WIN32 AND ONEMKL_SYCL_IMPLEMENTATION STREQUAL "dpc++")
```